app/services/stats/stats_report_generator.rb

```diff
@@ -29,24 +48,12 @@ def call
     attr_reader :report_type, :options
 
     def validate_report_type
-      raise InvalidReportType unless StatsReport::TYPES.include?(report_type.to_s)
+      raise InvalidReportType unless StatsReport.names.include?(report_type.to_s)
```

Comment:
The `StatsReport.names` method was part of a commit that you didn't include here so this needs to remain as `StatsReport::TYPES`.

```suggestion
      raise InvalidReportType unless StatsReport::TYPES.include?(report_type.to_s)
```

Reply:
done now